api/channel.go

```diff
@@ -0,0 +1,106 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/discordomo/api/router/middleware"
+	"github.com/gin-gonic/gin"
+)
+
+// Channel is the struct for channels
+type Channel struct {
+	Name string
+	Type string
+}
+
+// CreateChannel creates a channel
+func CreateChannel(c *gin.Context) {
+	channel := new(Channel)
+	err := c.Bind(channel)
+	if err != nil {
+		retErr := fmt.Errorf("Unable to parse json body: %w", err)
+		c.Error(retErr)
+		c.AbortWithStatusJSON(http.StatusBadRequest, retErr.Error())
+
+		return
+	}
+
+	dg := middleware.RetrieveSession(c)
+
+	cType := parseChannelType(channel.Type)
+	guildID, exists := os.LookupEnv("GUILD_ID")
+	if !exists {
+		retErr := fmt.Errorf("Guild ID not set")
+		c.Error(retErr)
+		c.AbortWithStatusJSON(http.StatusInternalServerError, retErr.Error())
+		return
+	}
+
+	_, err = dg.GuildChannelCreate(guildID, channel.Name, cType)
+	if err != nil {
+		return
```

Comment:
We should return an error to the user in this case.

Reply:
Error creating channel is now reported to the user.